napari/_qt/dialogs/qt_package_installer.py

```diff
@@ -0,0 +1,419 @@
+        combined_paths = os.pathsep.join(
+            [
+                user_site_packages(),
+                env.systemEnvironment().value("PYTHONPATH"),
+            ]
+        )
+        env.insert("PYTHONPATH", combined_paths)
+        env.insert("PIP_USER_AGENT_USER_DATA", _user_agent())
+        return env
+
+
+class CondaInstallerTask(AbstractInstallerTask):
+    def executable(self):
+        if self._executable:
+            return str(self._executable)
+        _bat = ".bat" if os.name == "nt" else ""
+        if exe := os.environ.get("MAMBA_EXE", shutil.which(f'mamba{_bat}')):
+            return exe
+        _exe = ".exe" if os.name == "nt" else ""
+        if exe := os.environ.get("CONDA_EXE", shutil.which(f'conda{_exe}')):
+            return exe
+        return 'conda'  # cross our fingers
+
+    def arguments(self) -> Tuple[str, ...]:
+        prefix = self.prefix or self._default_prefix()
+        args = [self.action, '-y', '--prefix', prefix]
+        args.append('--override-channels')
+        for channel in (*self.origins, *self._default_channels()):
+            args.extend(["-c", channel])
+        return (*args, *self.pkgs)
+
+    def environment(
+        self, env: QProcessEnvironment = None
+    ) -> QProcessEnvironment:
+        if env is None:
+            env = QProcessEnvironment.systemEnvironment()
+        PINNED = 'CONDA_PINNED_PACKAGES'
+        system_pins = f"&{env.value(PINNED)}" if env.contains(PINNED) else ""
+        env.insert(PINNED, f"napari={self._napari_pin()}{system_pins}")
+        if 10 <= log.getEffectiveLevel() < 30:  # DEBUG level
+            env.insert('CONDA_VERBOSITY', '3')
+        if os.name == "nt":
+            if not env.contains("TEMP"):
+                temp = gettempdir()
+                env.insert("TMP", temp)
+                env.insert("TEMP", temp)
+            if not env.contains("USERPROFILE"):
+                env.insert("HOME", os.path.expanduser("~"))
+                env.insert("USERPROFILE", os.path.expanduser("~"))
+        return env
+
+    def _napari_pin(self):
+        from ..._version import version, version_tuple
+
+        if "rc" in version or "dev" in version:
+            # dev or rc versions might not be available in public channels
+            # but only installed locally - if we try to pin those, mamba
+            # will fail to pin it because there's no record of that version
+            # in the remote index, only locally; to work around this bug
+            # we will have to pin to e.g. 0.4.* instead of 0.4.17.* for now
+            return ".".join([str(part) for part in version_tuple[:2]])
+        return ".".join([str(part) for part in version_tuple[:3]])
+
+    def _default_channels(self):
+        return ('conda-forge',)
+
+    def _default_prefix(self):
+        if (Path(sys.prefix) / "conda-meta").is_dir():
+            return sys.prefix
+        raise ValueError("prefix has not been specified!")
+
+
+class InstallerQueue(QProcess):
```

Comment:
If this is now just a Queu of qprocesses, then we should inherit from QObject only. Or is this just to preserve the base signals and structure of a QProcess? 🤔

Reply:
Technically it's a QProcess that consumes a Queue of "QProcess arguments", which are then used by the QProcess to configure itself. Is that an ok pattern or should we switch to a real Queue of QProcesses? We are using the signals quite a bit.

Reply:
I added some module level docs now.

Reply:
I think it is ok, just wanted to be sure. It is a Queue that has the same API as a QProcess... plus some extra signals, but we are not really using the `QProcess` class itself. 

For the sake of making sure it is used as intended I would probably prefer that we inherited from `QObject` and then recreated any signals as needed, but since this will be a more private internal detail for napari consuption only I would say it is ok for now :)

Reply:
Mmm the queue does use the QProcess functionality. It will set the command, arguments and environment variables using the instructions provided by the "Tool" (conda or pip). Then it will start itself. We can call it a Queue-aware QProcess?

Reply:
See `._process_queue()` for details.

Reply:
See `._process_queue()` for details.

Reply:
I prefer that this should be something inheriting from `QObject`. This is part of the installer that is not part of napari core and keeping this as `QObject` simplify keeping things in a proper thread.

Reply:
A different comment mentioned this should be made Qt agnostic, and I agree. But we will make that part of a later refactor, once we have extracted this to a separate package. Til then, this is part of napari core and it's blocking other work so I'd be inclined to keep it as is.